public/apps/configuration/panels/role-edit.tsx

```diff
@@ -47,24 +62,289 @@ const TITLE_TEXT_DICT = {
   create: 'Create Role',
   edit: 'Edit Role',
   duplicate: 'Duplicate Role',
-}
+};
 
 type OptionSeletion = EuiComboBoxOptionOption[];
+type FieldLevelSecurityMethod = 'exclude' | 'include';
+
+interface RoleIndexPermissionStateClass {
+  indexPatterns: OptionSeletion;
+  docLevelSecurity: string;
+  fieldLevelSecurityMethod: FieldLevelSecurityMethod;
+  fieldLevelSecurityFields: OptionSeletion;
+  maskedFields: OptionSeletion;
+  allowedActions: OptionSeletion;
+}
+
+function getEmptyIndexPermission(): RoleIndexPermissionStateClass {
+  return {
+    indexPatterns: [],
+    docLevelSecurity: '',
+    fieldLevelSecurityMethod: 'exclude',
+    fieldLevelSecurityFields: [],
+    maskedFields: [],
+    allowedActions: [],
```

Comment:
Since this is not named in a way similar to other attributes, probably move it after indexPatterns to follow the same order for ease of read. By the way, any reason why it is not named indexPermissions?

Reply:
Good idea.

Reply:
I was followed the attribute name that retrieved from server.

Reply:
If `indexPermissions` it will be also confusing with the higher level indexPermissions.